### PR TITLE
Adding dataFrame.printSchema

### PR DESCRIPTION
**Pull request by sfc-gh-smirzaei**
Overview: Adding `printSchema` method for printing a human readable version of dataFrame schema. Note that the jira asks for print_schema and printSchema(alias), but in the parent jira and pySpark we only have `printSchema`. I can make changes is we actually need the two methods mentioned in the jira ticket.

Precommit test link: https://ci-dev-142.int.snowflakecomputing.com/job/PythonStoredProcBuildPrecommitTest/545/

Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   SNOW-859941

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/snowpark/dataframe.py [hunk 1]
```diff
@@ -3771,6 +3771,16 @@ def convert(col: ColumnOrName) -> Expression:
         exprs = [convert(col) for col in parse_positional_args_to_list(*cols)]
         return exprs
 
+    def printSchema(self) -> str:
```
Comment (sfc-gh-aalam) [inline src/snowflake/snowpark/dataframe.py:3774]: `printSchema` is supposed to print the schema, not necessarily returning a string. We can use https://stackoverflow.com/questions/20507601/writing-a-pytest-function-for-checking-the-output-on-console-stdout to test the output.

I'm also thinking to keep all the functions consistent in snowpark, we name this function `print_schema` and alias `printSchema` for compatibility

File: src/snowflake/snowpark/dataframe.py [hunk 2]
```diff
@@ -3771,6 +3771,16 @@ def convert(col: ColumnOrName) -> Expression:
...
+        return "root\n%s" % (
```
Comment (sfc-gh-stan) [inline src/snowflake/snowpark/dataframe.py:3775]: NIT: use f-strings instead?

Comment (sfc-gh-aalam) [reply src/snowflake/snowpark/dataframe.py:3775]: I would second this. Looking at the failing tests, looks like we missed a `.format()`. We are moving away from `.format()` in favor of f-strings

File: tests/unit/test_dataframe.py [hunk 3]
```diff
@@ -273,3 +273,17 @@ def test_statement_params():
         param in kwargs["_statement_params"].items()
         for param in statement_params.items()
     )
+
+
+def test_dataFrame_printSchema(capfd):
+    mock_connection = mock.create_autospec(ServerConnection)
+    mock_connection._conn = mock.MagicMock()
+    session = snowflake.snowpark.session.Session(mock_connection)
+    session._conn._telemetry_client = mock.MagicMock()
```
Comment (sfc-gh-aalam) [inline tests/unit/test_dataframe.py:282]: I think you need to move this test to `tests/integ/test_dataframe.py`. Mock session is not helping to infer the schema of the input data.

File: src/snowflake/snowpark/dataframe.py [hunk 4]
```diff
@@ -3771,6 +3771,15 @@ def convert(col: ColumnOrName) -> Expression:
         exprs = [convert(col) for col in parse_positional_args_to_list(*cols)]
         return exprs
 
+    def print_schema(self) -> str:
+        schema_tmp_str = "\n".join(
```
Comment (sfc-gh-aalam) [inline src/snowflake/snowpark/dataframe.py:3775]:
```suggestion
    def print_schema(self) -> None:
        schema_tmp_str = "\n".join(
```

**Pull request merged**